### PR TITLE
Fix "Smart" Mobs Being Able To Open Reinforced  Doors

### PR DESCRIPTION
**Pull request by MrJeremyFisher**
Prevent mobs like Piglins and Villagers (Entities with the com.destroystokyo.paper.entity.Pathfinder#canOpenDoors property set to true) from opening doors, trapdoors, and fence gates if they are reinforced

### CODE REVIEW

File: plugins/citadel-paper/src/main/java/vg/civcraft/mc/citadel/listener/EntityListener.java [hunk 1]
```diff
@@ -1,30 +1,23 @@
 package vg.civcraft.mc.citadel.listener;
 
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.List;
-import java.util.UUID;
+import java.util.*;
```
Comment (Protonull) [inline plugins/citadel-paper/src/main/java/vg/civcraft/mc/citadel/listener/EntityListener.java:3]: Please don't do wildcard imports

Comment (MrJeremyFisher) [reply plugins/citadel-paper/src/main/java/vg/civcraft/mc/citadel/listener/EntityListener.java]: Thanks and apologies. Was probably an auto format.

File: plugins/citadel-paper/src/main/java/vg/civcraft/mc/citadel/listener/EntityListener.java [hunk 2]
```diff
@@ -4,27 +4,20 @@
...
 import org.bukkit.Material;
+import org.bukkit.Tag;
 import org.bukkit.block.Block;
 import org.bukkit.block.BlockFace;
-import org.bukkit.entity.Entity;
-import org.bukkit.entity.EntityType;
-import org.bukkit.entity.Hanging;
-import org.bukkit.entity.ItemFrame;
-import org.bukkit.entity.Player;
+import org.bukkit.entity.*;
```
Comment (Protonull) [inline plugins/citadel-paper/src/main/java/vg/civcraft/mc/citadel/listener/EntityListener.java:16]: Wildcard

File: plugins/citadel-paper/src/main/java/vg/civcraft/mc/citadel/listener/EntityListener.java [hunk 3]
```diff
@@ -4,27 +4,20 @@
...
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.EventPriority;
 import org.bukkit.event.Listener;
-import org.bukkit.event.entity.CreatureSpawnEvent;
-import org.bukkit.event.entity.EntityBreakDoorEvent;
-import org.bukkit.event.entity.EntityChangeBlockEvent;
-import org.bukkit.event.entity.EntityDamageByEntityEvent;
-import org.bukkit.event.entity.EntityExplodeEvent;
-import org.bukkit.event.entity.EntitySpawnEvent;
+import org.bukkit.event.entity.*;
```
Comment (Protonull) [inline plugins/citadel-paper/src/main/java/vg/civcraft/mc/citadel/listener/EntityListener.java:20]: Wildcard

**Pull request merged**